components/console_cmd_mqtt/console_mqtt.c

```diff
@@ -0,0 +1,445 @@
+        }
+
+        if ((mqtt_args.username->count > 0) && (mqtt_args.password->count > 0)) {
+            if (mqtt_args.uri->count > 0) {
+                mqtt_cfg = get_mqtt_config_username_password(mqtt_args.uri->sval[0],
+                           mqtt_args.username->sval[0],
+                           mqtt_args.password->sval[0]);
+            } else {
+                mqtt_cfg = get_mqtt_config_username_password(CONFIG_MQTT_BROKER_URL,
+                           mqtt_args.username->sval[0],
+                           mqtt_args.password->sval[0]);
+            }
+        } else {
+            if (mqtt_args.uri->count > 0) {
+                mqtt_cfg = get_mqtt_config_basic(mqtt_args.uri->sval[0]);
+            } else {
+                mqtt_cfg = get_mqtt_config_basic(CONFIG_MQTT_BROKER_URL);
+            }
+        }
+
+        ESP_LOGI(TAG, "broker: %s", mqtt_cfg.broker.address.uri);
+
+        if (mqtt_args.cafile->count > 0) {
+            if (server_cert_pem_start && server_cert_pem_end) {
+                mqtt_cfg.broker.verification.certificate = (const char *)server_cert_pem_start;
+            }
+        }
+
+        if (mqtt_args.cert->count > 0) {
+            if (client_cert_pem_start && client_cert_pem_end) {
+                mqtt_cfg.credentials.authentication.certificate = (const char *)client_cert_pem_start;
+            }
+
+            if ((mqtt_args.key->count > 0) && client_key_pem_start && client_key_pem_end) {
+                mqtt_cfg.credentials.authentication.key = (const char *)client_key_pem_start;
+            } else {
+                mqtt_cfg.credentials.authentication.key = NULL;
+            }
+        }
+
+        client_handle = esp_mqtt_client_init(&mqtt_cfg);
+        if (client_handle == NULL) {
+            ESP_LOGE(TAG, "ERROR: Client init");
+            return 1;
+        }
+
+        /* The last argument may be used to pass data to the event handler, in this example mqtt_event_handler */
+        esp_mqtt_client_register_event(client_handle, ESP_EVENT_ANY_ID, mqtt_event_handler, NULL);
+        esp_mqtt_client_start(client_handle);
+
+        mqtt_config_free(mqtt_cfg);
+
+    } else if (mqtt_args.disconnect->count > 0) {
+        ESP_LOGD(TAG, "Disconnect command received:");
+
+        if (client_handle == NULL) {
+            ESP_LOGE(TAG, "mqtt client not connected");
+            return 0;
+        }
+
+        if (esp_mqtt_client_stop(client_handle) != ESP_OK) {
+            ESP_LOGE(TAG, "Failed to stop mqtt client task");
+            return 1;
+        }
+
+        client_handle = NULL;
+        ESP_LOGI(TAG, "mqtt client disconnected");
+    }
+
+    return 0;
+}
+
+
+esp_err_t set_mqtt_certs(const uint8_t *client_cert_pem_start_i, const uint8_t *client_cert_pem_end_i,
+                         const uint8_t *client_key_pem_start_i, const uint8_t *client_key_pem_end_i,
+                         const uint8_t *server_cert_pem_start_i, const uint8_t *server_cert_pem_end_i)
```

Comment:
nitipick, I'd probably call these `ca_cert`... rather than `server_cert` (to be consistent)